salt/state.py

```diff
@@ -3383,11 +3417,17 @@ def _filter_matches(_match, _data, _opts):
                                 _tmpdata = item.pop('subfilter')
                                 for match, data in six.iteritems(_tmpdata):
                                     _filter_matches(match, data, _opts)
+                            # The item is a reference to a state.
                             if isinstance(item, six.string_types):
                                 matches[saltenv].append(item)
+                            # The item is either a keyword parameter
+                            # or a reference to an environment.
                             elif isinstance(item, dict):
                                 env_key, inc_sls = item.popitem()
                                 if env_key not in self.avail:
+                                    # Pass through any keyword parameters.
+                                    matches[saltenv].append({ env_key:
+                                                                  inc_sls })
```

Comment:
Apologies for asking this again, but I wonder if it'd would be all right to simplify this even further?

As far as I can tell, the documented syntax for the top file allows currently two possibilities under a match: a) a reference to a state (e.g. `- some.state` and b) a keyword parameter "match" (e.g. `- match: matchtype`.

If I understand the block of code under discussion correctly, it tries to handle an entry which refers to a state in another environment (e.g. `- someenv: some.state`). And I can't seem to find such a thing from the [documentation](https://docs.saltstack.com/en/latest/ref/states/top.html).

Thus, to me it appears, and I may have understood this incorrectly, it'd be best to simply prune out the unnecessary cross-environment magic entirely. That'd make the code _much_ simpler, because now we could just pass any dictionaries right through.